source/core/slang-uint-set.h

```diff
@@ -31,13 +81,16 @@ class UIntSet
         /// Return the count of all bits directly represented
     Int getCount() const { return Int(m_buffer.getCount()) * kElementSize; }
 
+    List<Element>& getBuffer() { return m_buffer; }
```

Comment:
I looked through the uses in the code: we need access to the backing buffer for 2 different forms of serialization which are within 2 different programs.

We could minimize access but we cannot fully remove.

Reply:
Can we at least return `const List<Element>` with `const`?

Reply:
we can return a `const List&`, although I don't think it means much since we use it for `getBuffer` to modify the pointer in the buffer

I suppose we can use `friend` to allow only the serializer's to access the `CapabilitySet` methods in question.

Reply:
Please stay away from `friend` as much as possible.
I personally thing that it is worse than making all member variables public, because it creates ugly dependencies.

Ideally, accessing the internal data should be done only by the iterator.
But this function resembles more of `std::vector::data()`, which is arguably a good thing.
It is tricky to get rid of but we should try not to use it as much, I guess.

Reply:
I found a way to remove all of this "rule breaking" code which was added. No need to add `friend`